### PR TITLE
Get last modified date for site.

### PR DESCRIPTION
**Pull request by nimkamp**

### CODE REVIEW

File: cypress/e2e/scrape.server.cy.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      .click({ force: true });
+
+    cy.get('[data-test="jsonDisplay"]')
+      .should("exist")
+      .invoke("text")
+      .then((jsonText) => {
+        const jsonData = JSON.parse(jsonText);
+        expect(jsonData.info.timeUpdatedSource).to.equal("archive.org");
+        const timeStampDate = new Date(jsonData.info.timeUpdated * 1000);
+        const twentyFourHoursAgo = new Date(Date.now() - 24 * 60 * 60 * 1000);
```
Comment (scottPlusPlus) [inline cypress/e2e/scrape.server.cy.ts:26]: nit: let's also add a test that the returned timeStamp is not too far in the future

File: app/code/scout/twitter.ts [hunk 2]
```diff
@@ -32,7 +32,48 @@ export async function getUserData(username: string) {
...
+  const authorName = user.name;
+  const description = user.description;
+  const profileImageUrl = user.profile_image_url;
+  const followersCount = user.public_metrics.followers_count;
+  console.log("Author name:", authorName);
+  console.log("Description:", description);
+  console.log("Profile image URL:", profileImageUrl);
+  console.log("Followers:", followersCount);
+
+  return {
```
Comment (scottPlusPlus) [inline app/code/scout/twitter.ts]: one main problem here:

we don't want to be replacing the main "scrapeData" summary, content type, image, etc. with information from twitter.

the ONLY data we want from twitter is the author name, time updated, time updated source, and follower count as likes.  Everything else should remain unchanged.

1. Update the twitter.fetchTwitterData to return only the data we need from this.
2. Add a check that the bearer token exists and if not, log a warning and return null
3. wrap it in a try-catch.  If the request to twitter fails log a warning and return null

4. the scrapePage file should be responsible for calling the twitter.fetchTwitterData, and if there is valid data updating the ScrapedInfo object it will return

File: app/code/archive.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
+const axios = require("axios");
+
+// export async function getArchiveOrgIdentifier(url: string): Promise<string> {
```
Comment (scottPlusPlus) [inline app/code/archive.ts]: delete all this if not needed

File: app/code/archive.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+//   // Return the archive.org identifier
+//   const archiveMetaData = data.archived_snapshots.closest.url.split("/");
+//   console.log("Array: ", archiveMetaData);
+//   archiveMetaData.pop();
+//   console.log("Array2: ", archiveMetaData);
+//   return archiveMetaData.pop();
+// }
+
+export async function getLatestSnapshot(url: string) {
```
Comment (scottPlusPlus) [inline app/code/archive.ts]: please specify return type.  (ensure it's nullable in case of failure)

File: app/code/archive.ts [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    });
+    const jsonData = response.data;
+    console.log("JSON Data: ", jsonData);
+    return jsonData.archived_snapshots.closest.timestamp;
+  } catch (error) {
+    console.log("Error:", error);
+  }
+}
+
+export function getUnixTimeStamp(gmtTime: any): any {
```
Comment (scottPlusPlus) [inline app/code/archive.ts]: getLatestSnapshot function is responsible to return the TIME of the latest snapshot in the expected format (unix timestamp)

1. rename to getLatestSnapshotTime
2. getLatestSnapshotTime should call getUnixTimeStamp
3. please update getUnixTimeStamp with proper type definitions, and add a comment with an example of the expected input string

File: app/code/scout/twitter.ts [hunk 6]
```diff
@@ -32,7 +32,48 @@ export async function getUserData(username: string) {
...
+  const getUserEndpoint =
+    "https://api.twitter.com/2/users/by?usernames=" + twitterUsername;
+  const getUserResponse1 = await fetch(getUserEndpoint, options);
+  const getUserResponseJson = await getUserResponse1.json();
+  const userId = getUserResponseJson.data[0].id;
+  const getTweetsEndpoint = `https://api.twitter.com/2/users/${userId}/tweets?tweet.fields=created_at&max_results=5`;
+  console.log("\n= = = = = SENDING TWITTER REQUEST = = = = = \n");
+  const getTweetsResponse = await fetch(getTweetsEndpoint, options);
+  const getTweetsData = await getTweetsResponse.json();
+  const lastTweet = getTweetsData.data[getTweetsData.data.length - 1];
```
Comment (scottPlusPlus) [inline app/code/scout/twitter.ts]: we want the LATEST tweet, aka the newest :P  I assume they come in the order of newest first?

File: app/routes/activists2.tsx [hunk 7]
```diff
@@ -80,7 +80,6 @@ export default function ActivistsPage() {
   const loadedItems = itemsFromRemixData(data.collectionItems, infoMap);
 
   return (
-
```
Comment (scottPlusPlus) [inline app/routes/activists2.tsx:83]: nit:  try to avoid any unnecessary changes

File: app/code/archive.ts (hunk 3 above)
Comment (nimkamp) [reply app/code/archive.ts]: Removed this code.

File: app/code/archive.ts (hunk 4 above)
Comment (nimkamp) [reply app/code/archive.ts]: Return type is now specified. We are now returning null in case of failure or if data does not exist.

File: app/code/archive.ts (hunk 5 above)
Comment (nimkamp) [reply app/code/archive.ts]: Made the necessary changes.

File: app/code/scout/twitter.ts (hunk 6 above)
Comment (nimkamp) [reply app/code/scout/twitter.ts]: Ah sorry about that. This is now fixed.

File: app/code/scout/twitter.ts (hunk 2 above)
Comment (nimkamp) [reply app/code/scout/twitter.ts]: Completed this.

File: cypress/e2e/scrape.server.cy.ts (hunk 1 above)
Comment (nimkamp) [reply cypress/e2e/scrape.server.cy.ts:26]: Just added the test. This is complete.

File: app/routes/activists2.tsx (hunk 7 above)
Comment (nimkamp) [reply app/routes/activists2.tsx:83]: Should I put the newline back?

Comment (nimkamp) [reply app/routes/activists2.tsx:83]: Added back in.

File: app/code/archive.ts [hunk 8]
```diff
@@ -45,19 +15,29 @@ export async function getLatestSnapshot(url: string) {
...
     const jsonData = response.data;
     console.log("JSON Data: ", jsonData);
-    return jsonData.archived_snapshots.closest.timestamp;
+    if (jsonData?.archived_snapshots?.closest?.timestamp) {
+      let lastModifiedTime = getUnixTimeStamp(
+        jsonData.archived_snapshots.closest.timestamp
+      );
+      return lastModifiedTime;
+    } else {
+      return undefined;
```
Comment (scottPlusPlus) [inline app/code/archive.ts]: nit: stick to null

File: cypress/e2e/scrape.server.cy.ts [hunk 9]
```diff
@@ -29,6 +29,34 @@ describe("getLatestSnapshot", () => {
       });
   });
 
+  it("should ensure that the return timestamp from wayback machine is not too far into the future", () => {
```
Comment (scottPlusPlus) [inline cypress/e2e/scrape.server.cy.ts]: nit: we can just include this  in the first test.  Get the one timeStampDate and then make two assertions (expect) against it.

File: app/code/scout/twitter.ts [hunk 10]
```diff
@@ -38,42 +42,45 @@ export async function getUserData(username: string) {
 }
 
 export async function fetchTwitterData(twitterUsername: string): Promise<any> {
```
Comment (scottPlusPlus) [inline app/code/scout/twitter.ts]: please define the return type.

File: app/code/scout/ScrapePage.server.ts [hunk 11]
```diff
@@ -122,57 +125,31 @@ async function scrapePageImpl(urlStr: string): Promise<ScrapedInfo> {
...
-      });
+    if (twitterUsername) {
+      let twitterObject;
+      try {
+        twitterObject = await twitter.fetchTwitterData(twitterUsername);
+      } catch (error) {
+        console.error("Failed to fetch Twitter data: ", error);
+      }
+      console.log("TWITTER OBJECT: ", twitterObject);
+      return twitterObject;
```
Comment (scottPlusPlus) [inline app/code/scout/ScrapePage.server.ts]: let's talk about this on Wednesday.  To think about:  what is the job / responsibility of this scrapePage function?  How does that relate to the info from twitter?

File: app/code/archive.ts (hunk 8 above)
Comment (nimkamp) [reply app/code/archive.ts]: resolved.

File: app/code/scout/twitter.ts (hunk 10 above)
Comment (nimkamp) [reply app/code/scout/twitter.ts]: Fixed.

File: cypress/e2e/scrape.server.cy.ts (hunk 9 above)
Comment (nimkamp) [reply cypress/e2e/scrape.server.cy.ts]: Yes we can. Fixed this.
